darts/utils/data/encoders.py

```diff
@@ -0,0 +1,570 @@
+        attribute
+            the attribute of the underlying pd.DatetimeIndex from  for which to add scalar information.
+            Must be an attribute of `pd.DatetimeIndex`, or `week` / `weekofyear` / `week_of_year` - e.g. "month",
+            "weekday", "day", "hour", "minute", "second". See all available attributes in
+            https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DatetimeIndex.html#pandas.DatetimeIndex.
+            For more information, check out :meth:`datetime_attribute_timeseries()
+            <darts.utils.timeseries_generation.datetime_attribute_timeseries>`
+        """
+        super(DatetimeAttributeFutureEncoder, self).__init__(
+            index_generator=FutureCovariateIndexGenerator(input_chunk_length, output_chunk_length),
+            attribute=attribute
+        )
+
+
+class PositionalEncoder(SingleEncoder):
+    """PLACEHOLDER: absolute positional index encoding"""
+
+    def __init__(self, index_generator, *args):
+        super(PositionalEncoder, self).__init__(index_generator)
+        raise_if(True, 'NotImplementedError')
+
+    def _encode(self, index: SupportedIndexes) -> TimeSeries:
+        super(PositionalEncoder, self)._encode(index)
+        return TimeSeries.from_times_and_values(index, np.arange(len(index)))
+
+
+class PositionalPastEncoder(PositionalEncoder):
+    """PLACEHOLDER: absolute index encoder for past covariates"""
+    def __init__(self, input_chunk_length, output_chunk_length, *args):
+        super(PositionalPastEncoder, self).__init__(
+            index_generator=PastCovariateIndexGenerator(input_chunk_length, output_chunk_length),
+            *args
+        )
+
+
+class PositionalFutureEncoder(PositionalEncoder):
+    """PLACEHOLDER: absolute index encoder for future covariates"""
+    def __init__(self, input_chunk_length, output_chunk_length, *args):
+        super(PositionalFutureEncoder, self).__init__(
+            index_generator=FutureCovariateIndexGenerator(input_chunk_length, output_chunk_length),
+            *args
+        )
+
+
+class SequenceEncoder(Encoder):
+    """A sequence encoder can store and control multiple past and future covariate encoders at once.
+    It provides the same functionality as single encoders (`encode_train()` and `encode_inference()`).
+    Sequence encoders can be used with Darts' darts `Datasets`.
+    """
+
+    def __init__(self,
+                 model_kwargs: Dict,
+                 input_chunk_length: int,
+                 output_chunk_length: int,
+                 shift: int,
+                 takes_past_covariates: bool = False,
+                 takes_future_covariates: bool = False) -> None:
+
+        """
+        SequenceEncoder automatically creates encoder objects from parameters `model_kwargs` used when creating a
+        `TorchForecastingModel`.
+
+        *   Only kwarg `add_encoders` of type `Optional[Dict]` will be used to extract the encoders.
```

Comment:
Was there a reason why taking all the model's kwargs and not just the `add_encoders` parameter here?

Reply:
True, we can directly use the `add_encoders` parameter.
I did it in the beginning to be consistent with the docs for the Model itself and the SequenceEncoder